tests/analysis/Decompilation/dynamic_native/array.c

```diff
@@ -0,0 +1,27 @@
+#include <stdint.h>
+
+struct __attribute__((packed)) X {
+  uint16_t a[10];
+  uint64_t b;
+};
+
+uint16_t __attribute__((noinline)) getA(struct X *x) {
+  uint16_t max = 0;
+  for (int i = 0; i < 10; i++) {
+    uint16_t cur = x->a[i];
+    if (cur > max)
+      max = cur;
+  }
+
+  return max;
+}
+
+uint64_t __attribute__((noinline)) getB(struct X *x) { return (x->b) * 2; }
+
+int __attribute__((noinline)) main(int argc, char **argv) {
+  struct X *TheX = *(struct X **)(argv[0]);
```

Comment:
Not super nice, can you maybe have a separate function that uses a global variable.
Would that make a difference?

Reply:
I can check how the code is generated and decompiled using global variables. I think that the simpler the decompiled code comes out from this, the better, since it is meant to be inspected by hand.

Reply:
I'm just suggesting to have a global variable of type `struct X *` and used that instead of using `argv[0]`.
You will see a load from a fixed address as opposed to using the argument (in the register).